arcgis-ios-sdk-samples/Layers/List KML contents/ListKMLContentsSceneViewController.swift

```diff
@@ -130,71 +130,84 @@ class ListKMLContentsSceneViewController: UIViewController {
-                print("Unexpected AGSKMLViewpointType \(kmlViewpoint.type)")
-                return nil
+            return viewpointFromKMLViewpoint(kmlViewpoint)
+            
+        } else if let extent = node.extent {
+            // the node does not have a predefined viewpoint, so create a viewpoint based on its extent
+            return viewpointFromKMLNodeExtent(extent)
+        }
+        // the node doesn't have a predefined viewpoint or geometry
+        return nil
+    }
+    
+    // Converts the KML viewpoint to a viewpoint for the scene.
+    // The KML viewpoint may not correspond to the node's geometry.
+    private func viewpointFromKMLViewpoint(_ kmlViewpoint: AGSKMLViewpoint) -> AGSViewpoint? {
+        
+        switch kmlViewpoint.type {
+        case .lookAt:
+            var lookAtPoint = kmlViewpoint.location
+            if kmlViewpoint.altitudeMode != .absolute {
+                // if the elevation is relative, account for the surface's elevation
+                let elevation = sceneSurfaceElevation(for: lookAtPoint) ?? 0
+                lookAtPoint = AGSPoint(x: lookAtPoint.x, y: lookAtPoint.y, z: lookAtPoint.z + elevation, spatialReference: lookAtPoint.spatialReference)
             }
+            let camera = AGSCamera(lookAt: lookAtPoint,
+                                   distance: kmlViewpoint.range,
+                                   heading: kmlViewpoint.heading,
+                                   pitch: kmlViewpoint.pitch,
+                                   roll: kmlViewpoint.roll)
+            // only the camera parameter is used by the scene
+            return AGSViewpoint(center: kmlViewpoint.location, scale: 1, camera: camera)
+        case .camera:
+            // convert the KML viewpoint to a camera
+            let camera = AGSCamera(location: kmlViewpoint.location,
+                                   heading: kmlViewpoint.heading,
+                                   pitch: kmlViewpoint.pitch,
+                                   roll: kmlViewpoint.roll)
+            // only the camera parameter is used by the scene
+            return AGSViewpoint(center: kmlViewpoint.location, scale: 1, camera: camera)
+        case .unknown:
+            print("Unexpected AGSKMLViewpointType \(kmlViewpoint.type)")
+            return nil
         }
-            // the node does not have a predefined viewpoint, so create a viewpoint based on its extent
-        else if let extent = node.extent,
-            // some nodes do not include a geometry, so check that the extent isn't empty
-            !extent.isEmpty {
+    }
+    
+    /// Creates a default viewpoint framing the node based on its extent.
+    private func viewpointFromKMLNodeExtent(_ extent: AGSEnvelope) -> AGSViewpoint? {
+        
+        // some nodes do not include a geometry, so check that the extent isn't empty
+        guard !extent.isEmpty else {
+            return nil
+        }
+        
+        var center = extent.center
```

Comment:
I would suggest renaming this to `extentCenter` and making it immutable:

```swift
let extentCenter = extent.center
// take the scene's elevation into account
let elevation = sceneSurfaceElevation(for: extentCenter) ?? 0
```

Then declaring `center` as a new immutable variable inside the `if` clause:

```swift
let center = AGSPoint(x: extentCenter.x, y: extentCenter.y, z: extentCenter.z + elevation, spatialReference: extent.spatialReference)
// Defaults based on Google Earth.
let camera = AGSCamera(lookAt: center, distance: 1000, heading: 0, pitch: 45, roll: 0)
```